### DISCUSSION TITLE
How do I write Pixels outside of Quantum Range?

### DISCUSSION THREAD

**Discussion by Raildex** [dlemstra/Magick.NET#1452] [Help]
I have the Q16 HDRI version.

I am writing to pixels manually, like so:
```
var pixels = lightmap.GetPixels();
var px = pixels[pixelX,pixelY];
px[0] += value.X * Quantum.Max;
px[1] += value.Y * Quantum.Max;
px[2] += value.Z * Quantum.Max;
px[3] += amount * Quantum.Max;

```
where `value` is a float3 in the range of 0.0 to 1.0 (or slightly beyond that)

When writing and reading the image, I noticed that my values get clamped to 0 and 65535.0f

How can I work like any other image tool where i can add color values without clamping, Especially with the Exr format???

An image that has the color value R=4.0 G=0.2 B=1.0 should have R=4.0, G=0.2 and B=1.0 when writing and reading.

**Comment by dlemstra** [comment]
Our internal range is 0 to 65535 so your values will be changed to that range when you read an image. And for some formats this will be changed to a different range when it is being written. There is nothing you can do about that.
